numpy/core/src/umath/ufunc_object.c

```diff
@@ -1847,8 +1843,11 @@ _parse_axes_arg(PyUFuncObject *ufunc, int op_core_num_dims[], PyObject *axes,
             }
         }
         else {
-            PyErr_Format(PyExc_TypeError, "axes item %d should be a tuple",
-                         iop);
+            PyErr_Format(PyExc_TypeError,
+                    "%s: operand %d has %d core dimensions, "
+                    "but the axes item is not tuple.  (A single integer is "
+                    "also allowed when only one dimension is required.)",
```

Comment:
I think I'd leave out the parenthetical sentence here, or write it as "A single integer is only allowed when just one dimension is required" -- but I think it does not necessarily help, and people should just read the documentation if confused.

Reply:
Yeah, agreed. Removed.  Something doesn't quite feel ideal still, but not sure what :), and I think its an improvement either way.